catalog/registry/catalog-registry-api-impl/src/test/java/ddf/catalog/registry/api/TestRegistryStore.java

```diff
@@ -0,0 +1,211 @@
+        registryStore.setPushAllowed(false);
+        UpdateRequest request = new UpdateRequestImpl("testId", getDefaultMetacard());
+        registryStore.update(request);
+
+    }
+
+    @Test(expected = IngestException.class)
+    public void testDeleteDisabled() throws Exception {
+        registryStore.setPushAllowed(false);
+        DeleteRequest request = new DeleteRequestImpl("testId");
+        registryStore.delete(request);
+
+    }
+
+    @Test(expected = UnsupportedQueryException.class)
+    public void testQueryDisabled() throws Exception {
+        registryStore.setPullAllowed(false);
+        QueryRequest request =
+                new QueryRequestImpl(new QueryImpl(new GeotoolsFilterBuilder().attribute("anything")
+                        .is()
+                        .text("something")));
+        registryStore.query(request);
+
+    }
+
+    @Test
+    public void testUpdate() throws Exception {
+
+        registryStore = new RegistryStoreImpl(context, configuration, provider, factory) {
+            @Override
+            public SourceResponse query(QueryRequest queryRequest)
+                    throws UnsupportedQueryException {
+                List<Result> results = new ArrayList<>();
+                results.add(new ResultImpl(getDefaultMetacard()));
+                return new SourceResponseImpl(queryRequest, results);
+            }
+        };
+        registryStore.setParser(parser);
+        registryStore.setFilterBuilder(new GeotoolsFilterBuilder());
+        registryStore.setSchemaTransformerManager(transformer);
+
+        Csw csw = mock(Csw.class);
+        TransactionResponseType responseType = mock(TransactionResponseType.class);
+        TransactionSummaryType tst = new TransactionSummaryType();
+        tst.setTotalUpdated(new BigInteger("1"));
+        when(responseType.getTransactionSummary()).thenReturn(tst);
+        when(factory.getClientForSubject(any())).thenReturn(csw);
+        when(csw.transaction(any())).thenReturn(responseType);
+        when(transformer.getTransformerIdForSchema(any())).thenReturn(null);
+        when(configuration.getSchemaToTagsMapping()).thenReturn(new HashMap<>());
+        UpdateRequestImpl request = new UpdateRequestImpl("testId", getDefaultMetacard());
+        MetacardImpl updatedMcard = getDefaultMetacard();
+        updatedMcard.setId("newTestId");
+        OperationTransactionImpl opTrans =
+                new OperationTransactionImpl(OperationTransaction.OperationType.UPDATE,
+                        Collections.singletonList(updatedMcard));
+        request.getProperties()
+                .put(Constants.OPERATION_TRANSACTION_KEY, opTrans);
+        registryStore.update(request);
+        assertThat(request.getUpdates()
+                .get(0)
+                .getValue()
+                .getMetadata()
+                .contains("value=\"newTestId\""), is(true));
+    }
+
+    private void setParser(Parser parser) {
+
+        this.configurator =
```

Comment:
It seems strange that only a single Configurator is created where for the parser, but in the class 2 are created. Are both actually different and necessary?